walkthroughs/howto-k8s-retry-policy/deploy.sh

```diff
@@ -26,29 +26,54 @@ ECR_IMAGE_PREFIX="${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/
 FRONT_APP_IMAGE="${ECR_IMAGE_PREFIX}/feapp"
 COLOR_APP_IMAGE="${ECR_IMAGE_PREFIX}/colorapp"
 
+MANIFEST_VERSION="${1:-v1beta2}"
+
 error() {
     echo $1
     exit 1
 }
 
-check_appmesh_k8s() {
+check_k8s_virtualrouter() {
+    #check CRD
+    crd=$(kubectl get crd virtualrouters.appmesh.k8s.aws -o json )
+    if [ -z "$crd" ]; then
+        error "$PROJECT_NAME requires virtualrouters.appmesh.k8s.aws CRD to support retryPolicy. See https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/CHANGELOG.md"
+    else
+        echo "CRD check passed!"
+    fi
+}
+
+check_k8s_virtualservice() {
     #check CRD
     crd=$(kubectl get crd virtualservices.appmesh.k8s.aws -o json | jq -e '.. | .retryPolicy? | select(. != null)')
     if [ "$crd" = "" ]; then 
         echo "$PROJECT_NAME requires CRD that supports retryPolicy. See https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/CHANGELOG.md#v020"
         exit 1
     fi
+}
 
+check_appmesh_k8s() {
     #check aws-app-mesh-controller version
-    currentver=$(kubectl get deployment -n appmesh-system appmesh-controller -o json | jq -r ".spec.template.spec.containers[].image" | cut -f2 -d ':')
-    requiredver="v0.2.0"
+    currentver=$(kubectl get deployment -n appmesh-system appmesh-controller-manager -o json | jq -r ".spec.template.spec.containers[].image" | cut -f2 -d ':')
```

Comment:
We will have to get the currentver separately for v1beta2 and v1beta1 since we want to use a new deployment name, and the v1beta2 deployment has now contains two containers - kube-rbac-proxy and manager.

Reply:
Thanks for the review. That's a good point - I'll address it everywhere as a follow up